.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,109 @@
+# This workflow is a mix of:
+# - https://github.com/pola-rs/r-polars/blob/main/.github/workflows/check.yaml
+# - https://github.com/r-lib/actions/blob/v2-branch/examples/pkgdown.yaml
+# - https://squidfunk.github.io/mkdocs-material/publishing-your-site/?h=deploy#material-for-mkdocs
+
+name: mkdocs
+
+on:
+  push:
+    branches: main
```

Comment:
This will always push from the main branch.
pkgdown automatically determines whether to update the root or the `dev` directory by the version number of the R package, but this trigger may be inappropriate if there is no such mechanism.

Reply:
To me this means that the CI will run when there's a push on `main`, isn't that what we want? Also for now I don't think it's possible to have versioned documentation. I think `mkdocs` allows this but I didn't look into it and maybe it should be done in another PR later

Reply:
I don't think it is necessary to necessarily keep multiple versions of a document.

@sorhawell Do you have any idea which version of the document you would like to publish?
I think a while ago only tags would have triggered document updates.

Reply:
Before I only pushed docs for every binary release, but I think that was too rare. I think it makes sense to push docs at every push to main. At a later stage we could freeze the docs for each released version, similar to rust docs, but that is a luxury feature I think. Can a user who install polars from X commit build the docs herself locally?

Reply:
Yes they would only need to run `altdoc::update_docs(custom_reference = "docs/make-docs.R")` (at the condition that they have everything needed to build the package).

But the [docs versioning](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/?h=versioni) is also a feature of `mkdocs`, we could enable it later

Reply:
I agree that it is not really necessary to keep the documentation for the latest release since this package is not yet a CRAN release.

> they would only need to run `altdoc::update_docs(custom_reference = "docs/make-docs.R")` (at the condition that they have everything needed to build the package).

Perhaps it would be worthwhile to manage jobs on this using Makefile, etc. I have created an issue #173

Reply:
I never used Makefiles so I don't really understand what it would bring here. Is it to run `devtools::document` + `devtools::check` + `altdoc::update_docs(custom_reference = "docs/make-docs.R")` all at once? If so, how do you run it locally?

Reply:
You just type `make` in a bash/zsh terminal. Once you are ready, just post different code blocks with the different workflows that we need (series of commands) in the Makefile Issue, and one of us (I can) will write one.